xla/service/gpu/nccl_api.h

```diff
@@ -33,6 +33,21 @@ limitations under the License.
 
 namespace xla::gpu {
 
+//===----------------------------------------------------------------------===//
+// NcclCommConfig
+//===----------------------------------------------------------------------===//
+// NcclCommConfig contains configurations to initialize a communicator.
+// It wraps the ncclConfig_t object. More configuration parameters
+// will be exposed in the future when necessary.
+
+struct NcclCommConfig {
+  static absl::StatusOr<NcclCommConfig> Create(int64_t max_nchannels);
```

Comment:
Looks like a simple struct would work fine here? Can be converted to factory later if needed.

Reply:
sure, i will remove the factory for now.